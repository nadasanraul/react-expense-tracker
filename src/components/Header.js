import React from 'react';
import {NavLink} from 'react-router-dom';

const Header = () => (
    <nav className="navbar navbar-expand-lg navbar-dark bg-primary mb-5">
        <div className="container">
            <NavLink to="/" className="navbar-brand">Expenses Tracker</NavLink>
            <button className="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarCollapse" aria-controls="navbarCollapse" aria-expanded="false" aria-label="Toggle navigation">
                  <span className="navbar-toggler-icon"></span>
            </button>
            <div className="collapse navbar-collapse" id="navbarCollapse">
                <ul className="navbar-nav mr-auto ml-3">
                    <li className="nav-item">
                        <NavLink to="/" className="nav-link" activeClassName="active" exact={true}>Home</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink to="/create" className="nav-link" activeClassName="active">Add Expense</NavLink>
                    </li>
                    <li className="nav-item">
                        <NavLink to="/help" className="nav-link" activeClassName="active">Help</NavLink>
                    </li>
                </ul>
            </div>
        </div>
    </nav>
);

export default Header;