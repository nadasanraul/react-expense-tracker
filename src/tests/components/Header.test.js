import React from 'react';
import {shallow} from 'enzyme';
import Header from '../../components/Header';

test('should render Header correctly', () => {
    const wrapper = shallow(<Header />);
    expect(wrapper).toMatchSnapshot();
    
    // expect(wrapper.find('.navbar-brand').text()).toBe('Expenses Tracker');
    // const renderer = new ReactShallowRender();
    // renderer.render(<Header />);
    // expect(renderer.getRenderOutput()).toMatchSnapshot();
});