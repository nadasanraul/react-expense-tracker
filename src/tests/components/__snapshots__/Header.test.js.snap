// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header correctly 1`] = `
<nav
  className="navbar navbar-expand-lg navbar-dark bg-primary mb-5"
>
  <div
    className="container"
  >
    <NavLink
      activeClassName="active"
      ariaCurrent="true"
      className="navbar-brand"
      to="/"
    >
      Expenses Tracker
    </NavLink>
    <button
      aria-controls="navbarCollapse"
      aria-expanded="false"
      aria-label="Toggle navigation"
      className="navbar-toggler"
      data-target="#navbarCollapse"
      data-toggle="collapse"
      type="button"
    >
      <span
        className="navbar-toggler-icon"
      />
    </button>
    <div
      className="collapse navbar-collapse"
      id="navbarCollapse"
    >
      <ul
        className="navbar-nav mr-auto ml-3"
      >
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            exact={true}
            to="/"
          >
            Home
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/create"
          >
            Add Expense
          </NavLink>
        </li>
        <li
          className="nav-item"
        >
          <NavLink
            activeClassName="active"
            ariaCurrent="true"
            className="nav-link"
            to="/help"
          >
            Help
          </NavLink>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;
